CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/utilities/UpdateElementBitmaskPlugin.java

```diff
@@ -65,8 +65,8 @@ private void setVertices(final GraphWriteMethods graph, final int currentBitmask
     }
 
     private void setTransactions(final GraphWriteMethods graph, final int currentBitmask) {
-        final int transactionSelectedAttributeId = VisualConcept.TransactionAttribute.SELECTED.get(graph);
-        final int transactionBitmaskAttributeId = VisualConcept.TransactionAttribute.LAYER_MASK.get(graph);
+        final int transactionSelectedAttributeId = VisualConcept.TransactionAttribute.SELECTED.ensure(graph);
+        final int transactionBitmaskAttributeId = VisualConcept.TransactionAttribute.LAYER_MASK.ensure(graph);
         if (transactionSelectedAttributeId != Graph.NOT_FOUND) {
```

Comment:
You don't need to use ensure if you're then checking for `Graph.NOT_FOUND`. the ensure method will create the attribute so that it can never be "not found" - i'd stick with get here.

Reply:
My thoughts were that if .get() was called, and it did not exist - then the code within the block would not run. Meaning the user would try adding elements of the graph to a layer and it would appear as nothing was added.

Adding ensure means that when that plugin is ran, it will 100% be added to the layer on first try. I can change this @cygnus-x-1 if the above reasoning doesn't change things.

Reply:
I missed that this was in a plugin and not `StoreGraph` - so if your intention is for these attributes to always exist, then using `ensure` is correct. However that means you don't need the checks for `Graph.NOT_FOUND`.

Reply:
Sound good to me